src/web/app/pages-student/student-course-details-page/student-course-details-page.component.ts

```diff
@@ -1,46 +1,18 @@
 import { Component, OnInit } from '@angular/core';
 import { ActivatedRoute } from '@angular/router';
 import { environment } from '../../../environments/environment';
+import { CourseService } from '../../../services/course.service';
 import { HttpRequestService } from '../../../services/http-request.service';
 import { StatusMessageService } from '../../../services/status-message.service';
+import { StudentProfileService } from '../../../services/student-profile.service';
+import { StudentService } from '../../../services/student.service';
+import { Course, Instructors, Student, Students } from '../../../types/api-output';
 import { Gender } from '../../../types/gender';
 import { ErrorMessageOutput } from '../../error-message-output';
+import { StudentProfile } from '../../pages-instructor/student-profile/student-profile';
 
-interface StudentAttributes {
-  email: string;
-  course: string;
-  name: string;
-  googleId: string;
-  team: string;
-  section: string;
-}
-
-interface InstructorDetails {
-  name: string;
-  email: string;
-}
-
-interface CourseAttributes {
-  createdAt: string;
-  id: string;
-  name: string;
-  timeZone: string;
-}
-
-interface TeammateProfile {
-  shortName: string;
-  email: string;
-  institute: string;
-  nationality: string;
-  gender: Gender;
-  pictureKey: string;
-}
-
-interface StudentCourseDetails {
-  student: StudentAttributes;
-  course: CourseAttributes;
-  instructorDetails: InstructorDetails[];
-  teammateProfiles?: TeammateProfile[];
+interface StudentProfileWithPicture extends StudentProfile {
```

Comment:
Prefer composition over inheritance. i.e. It is better to put student profile as a field inside the interface rather than extending it.

Reply:
Yupp changed